### ISSUE TITLE
[BUG] MK4 keeps requesting filament path to be cleared upon dried spool change

### ISSUE THREAD

**Issue by bolsoncerrado** [prusa3d/Prusa-Firmware-Buddy#3129]
Please, before you create a new bug report, please make sure you searched in open and closed issues and couldn't find anything that matches.

**Printer type** - [MINI]

**MK4**

**Printer firmware version** - [e.g. 4.0.5, ...]

4.5.4

**Original or Custom firmware** - [Original/Custom]

Original

**Optional upgrades** - [e.g. Filament Runout Sensor]

Stock Factory Assembled MK4

**USB drive or USB/Octoprint**
  Please let us know if you print via USB flash drive or USB/Octoprint

USB

**Describe the bug**

1. Start a print
2. Spool runs out of filament
3. Filament sensor kicks in
4. Filament is removed BUT
5. even if you reply YES to the "was filament removed ok?" the printer KEEPS asking you to remove filament. Nothing helps, the dial, the clicky, nothing. Only a RESET, ruining the print.
6. Upon rebooting filament sensor works as intended, so there's no debris no filament messing the sensor, no nothing.

**How to reproduce**

Read above.

**Expected behavior**

Since it has happened to me already TWICE and non correlative, I asume this is a firmware glitch. I already blowed air, moved filament on the port hole, nothing helps when the bug is present. ONLY Reset button fix the issue and once the printer reboots, everything works as intended.

**G-code**
 
No particular gcode.

**Video**

Next time it happens I'll tape it.

**Comment by bolsoncerrado** [comment]
@marcelismus what fw version did u have issues with? mine is 4.6.1 -- i'll install alpha3 now.

**Comment by unclej84** [comment]
I am having the same issue with 4.7.1 firmware - goes to park position, unloads and shows confirm screen (selected "yes") but still shows to clear the path

**Comment by unclej84** [comment]
Made a test with colorchange today and there the filament change worked - don't know what the difference between both cases (color change <-> filament runout) internally is

**Comment by bolsoncerrado** [comment]
Idk either but I lost another print today again, pretty frustrating.

**Comment by unclej84** [comment]
@bolsoncerrado are you on alpha3 now or still on 4.6.1?

**Comment by bolsoncerrado** [comment]
Now on 4.7

**Comment by arcenik86** [comment]
Confirmed issue on 4.7.1, however I did get it to clear after inserting and removing filament quickly a few times. Before doing so, verified there was no filament in the idler area. New MK4 build, third print.

**Comment by bolsoncerrado** [comment]
I tried that technique and it didnt work for me :(

**Comment by IftiBashir** [comment]
I would consider this a rather serious issue. Have Prusa responded?

**Comment by bolsoncerrado** [comment]
No :(

**Comment by zhiyao** [comment]
It happened to me as well twice. 

I realized the filament sensor is not calibrated for me and will always register as a detected filament.

Temp Fix: The only way to fix this is to do a control > filament sensor calibration.

Proposed Fix: There should have been a way to overwrite the sensor during such an issue.

**Comment by unclej84** [comment]
Firmware 4.7.2 got released. Has anyone already tested, if it is fixed? Release note does not mention anything related.

**Comment by bolsoncerrado** [comment]
Why would they? No one from Prusa ever commented here :(

**Comment by MilleniumDawn** [comment]
Same issue on v5.0.0-alpha4. Stuck on requesting filament path to be cleared.
I'll give 4.7.2 a try...

**Comment by derthomers** [comment]
same here on 5.0.0-alpha4. pushing in and pulling out a piece of filament multiple times worked in some cases. had it 8 times in 2 weeks. also on 5.0.0-alpha1..3. that really tends to ruin your print if you want to empty a spool completely before switching to a new one.

**Comment by bolsoncerrado** [comment]
Happens also if you plan to leave and don't trust enough on the filament sensor stopping the print hence if you manually opt to change filament sensor may ruin the print awaiting you remove the filament you've already acknowledged to have removed correctly.

Pretty absurd altogheter.

**Comment by arcenik86** [comment]
Just wanted to comment that this is easily duplicated with a filament change from the printer menu as well. Even after you confirm the filament is removed (yes on the prompt), you have to toggle the filament in and out for the printer to recognize nothing is loaded.

**Comment by bolsoncerrado** [comment]
What I just said yep.

**Comment by micro17** [comment]
Had the same odd behaviour today. Would be nice if Prusa can response if this is a software or hardware bug.

For me it cleared after putting the filament in and out multiple times

**Comment by bolsoncerrado** [comment]
SOunds to me they have a severe hardware design issue with this filament sensor placement :(

This radio silence is nothing but worrysome!

I also wonder if XL people is having the same issue...

**Comment by bolsoncerrado** [comment]
Well it's quite absurd the screen asks you "was the filament removed correctly?" just to ask you again to please remove the filament from the sensor path when you can't clearly reach it AND no other option (specially mid print) left than pray that wiggling multiple times the new filament will trigger the sensor :(

So I think it's both a hardware design flaw (like having a hole that gravity will end up filling with dust and such anyway) and a firmware issue...

**Comment by FliesWithWind** [comment]
+1 on this. It just happened to me when I runed out of filament. Wasted 24h of print....

**Comment by HaWiWe** [comment]
I had this happen a few times and could fix it by sticking the filament in and pulling out again a few times. It seems the hall sensor does not spring back, you can notice that there is no resistance when doing the above. At least that's what I think is happening.

**Comment by unclej84** [comment]
In the Podcast they just mentioned that new final firmware with input shaping will be released (around) Friday - let's see if this included

**Comment by Shushuda** [comment]
I just had this issue on 5.0.0-RC. The ball didn't spring back, there was no resistance. Inserting and pulling out a looot of times very rapidly finally made it spring back, resistance was there and the printer started detecting filament correctly.

**Comment by bolsoncerrado** [comment]
I think all these posts prove they have a serious hw design issue...

**Comment by Nulifier** [comment]
Something that worked for me to prevent the print from failing is to bypass the filament sensor when it gets into this stuck state.

What I did after receiving the remove the filament from the filament sensor message:

1. Remove the LoveBoard-cover-right using the M3 screw on the right side of the extruder. [(Link)](https://help.prusa3d.com/guide/5-nextruder-assembly_434014#435352)
2. Unplug the filament sensor cable from the lower connector. I found a pair of tweezers helpful. [(Link)](https://help.prusa3d.com/guide/5-nextruder-assembly_434014#435214)
3. The message should clear and you should be left with the normal load filament message
4. Install the filament sensor cable again
5. Reinstall the LoveBoard-cover-right (Its important to do it now as once you load the filament the extruder will start zipping around)
6. Load the filament

Obviously this involves some delicate work in the middle of a print and take your own precautions but it did allow me to resume a 6 hour print after hour 4. But definitely needs to be fixed on a hardware or software level.

**Comment by bolsoncerrado** [comment]
Perhaps a cleaner, less risky way would be to add some small reset button somewhere to cut the power just enough for the message to clear....

**Comment by bolsoncerrado** [comment]
Perhaps a savaged mechanical endstop? 😂

**Comment by bolsoncerrado** [comment]
If anyone has links or names or part numbers to the male and female connectors may be I can order some and test a bypass system for this until a solution comes out. Granted this issue was opened on July 7th and no one from Prusa has chimmed in & granted we have had 2 firmware revisions in between i think it will still take a while for us to see a feasible solution and not just a temp fw patch...

**Comment by Nulifier** [comment]
> If anyone has links or names or part numbers to the male and female connectors 

Pretty sure this is a [Molex 5025850370](https://www.digikey.com/en/products/detail/molex/5025850370/2819043)

**Comment by bolsoncerrado** [comment]
Cool, what about the male One? :)

**Comment by bolsoncerrado** [comment]
This? https://www.digikey.es/en/products/detail/molex/5025780300/2818999

**Comment by Prusa-Support** [comment]
Thanks for reaching out to us.

This seems to be a duplicate of https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3056.
I'd rather close the first issue in favor of this one, as this duplicate grew in popularity.

We acknowledge the possible presence of a problem based on the number of reports on this issue.
However unfortunately we have not been able to confirm the presence of a bug so far.

We are working on this. If you encounter this issue again though, please follow these steps before reporting.

- Re-inserting the filament several times may knock off debris from the filament sensor mechanism.
- Find the cleaning hole on the top of the Nextruder, in the middle of it, next to the brass ring, and blow canned air through it.
- As the last resort, you can remove the "Nextruder sandwich" (brass ring + plastic parts, so you don't need to repeat the PG alignment) and remove the _filament sensor spring_ for deeper inspection and cleaning.

Also, make sure you have tested the most recent firmware version.
Consider testing the non-final versions available on [GitHub](https://github.com/prusa3d/Prusa-Firmware-Buddy/releases) if you don't mind (could be unstable).

**Please refrain from disconnecting components or touching electronic components while the printer is powered on as this may cause damages.**

Michele Moramarco
Prusa Research

**Comment by bolsoncerrado** [comment]
Wiggling constantly is the only thing that worked for me so far, as long  as I have patience and I don't move the print head accidentally during the process.... ::

**Comment by bolsoncerrado** [comment]
And btw I think the "bug" is actually a hw design flaw as one of my 2 mk4 started failing almost "next day", so no many prints on it... :/

**Comment by mataksik** [comment]
5.0.0RC same problem. Helped only disconecting filament sensor. 7h print was ruined

**Comment by bolsoncerrado** [comment]
5.0.0 FINAL IS version: 

Not only I keep having the same old issue that now I have a NEW one too:

Printer requests to change filament when filament is still present and printing!! :( :/ :( 

![20230920_110531](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/10992914/5fac1d92-2f40-44df-947d-ef869d58d0db)

**Comment by enerbe** [comment]
I can reproduce it. Made a video. https://youtu.be/0cFO8x1V8tU

It is not even needed to be in the middle of a print. The video contains the process to reproduce.

**Comment by bolsoncerrado** [comment]
Not the same last issue I reported here but yeah, pretty much no way out.... Unless you violently start moving up, down and dragging the filament on the sides until the sensor realizes you're moving it....


But having to reload filament midprint because the printer feels so is just meeeeh

**Comment by ladislav37** [comment]
We have new MK4 (with 5.0.0 fw ver.) and have the same issue.. It looks like this is not a fw/sw problem but hardware... Most probably spring force is too low and it will not push the ball to correct place for filament detection. But this is only my assumption, because I can't find online any schematics or models for MK4 and I don't want to disassemble printer extruder right now.. :-)

**Comment by bolsoncerrado** [comment]
Wdy mean schematics or models? To do what?

I guess you can take this as reference and perhaps source a stronger spring hmmm

https://help.prusa3d.com/guide/5-nextruder-assembly_434014

**Comment by bolsoncerrado** [comment]
Plus the fact there's an opening on the extruder housing and any dust or buildup or even filament particles when pulling out the filament may collapse the system... IMHO the whole thing is flawed.

You may have issues if you put more pressure thru a new spring and the actual issue is ball does not pull back.

**Comment by JonasFischer1** [comment]
I currently have the same issues on the MK4. During printing the filament sensor stops the print at random times and only allows printing again after a reset. 

The only thing that worked for me was to turn off the filament sensor in the settings menu.

**Comment by eklynx** [comment]
I just ran in to this as well, changing filament mid-print due to spool running out, but it wouldn't recognize the filament was removed and i had to reset/abort.

**Comment by bolsoncerrado** [comment]
Next time instead of reset  try inserting and removing and wiggling the filament quickly until sensor triggers...

**Comment by eklynx** [comment]
Already tried that. Didn't help

**Comment by unclej84** [comment]
@Prusa-Support Did I get the last Paragraph under "Current problems of MMU3 on the MK4" in the recent [blog post](https://blog.prusa3d.com/development-diary-update-on-mmu3-for-the-mk4_85043/) right, that you are also referring to this issue?

**Comment by Prusa-Support** [comment]
The post on the blog kind of confirms these anomalies are most likely related to debris and irregularly shaped filament that may cause a mechanical blockage of the filament sensor moving parts.

As you probably already know now, they are working on improving the filament tip shaping through a sequence of movements. That is a work in progress but theoretically will drastically reduce the chances of filament sensing problems.

That mostly concerns Multi Material prints where several hundreds of filament changes may occur in a single print.
As of now, while we work on the best filament shaping, our best recommendation is to clean the filament sensor.
https://help.prusa3d.com/article/filament-sensor-mk4-mk3-9_465702
Nonetheless, sometimes reinserting the filament a few times can do the trick.

Michele Moramarco
Prusa Research

**Comment by bolsoncerrado** [comment]
With all due respect these issues have been happening since day 0 way before mmu3 was even released :/

But ok. We'll take your word.

**Comment by IftiBashir** [comment]
The issues have been happening since the beginning, but the seriousness has only been highlighted now since it causes huge issues when using the MMU3. Otherwise I wonder how long these issues would have been apparent before Prusa worked on a fix. In my opinion, this 'fix' with the change in filament tip should be implemented asap - MMU3 or no MMU3.

**Comment by bolsoncerrado** [comment]
And TBH this filament tip thing is just a "patch" to a meh sensor configuration...

**Comment by jakkaj** [comment]
Happening to me too now. Can't even get the calibration steps now after printer reset  :/ 5.0.1 firmware.



![signal-2023-11-26-100201_002](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/5225782/1e80c6ea-b30f-45cd-97c7-ff07115c8542)
![signal-2023-11-26-100201_003](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/5225782/055dbf27-f121-42fc-a21d-1bcb1ff7c237)

**Comment by jakkaj** [comment]
The cleaning instructions here did help: https://help.prusa3d.com/article/filament-sensor-mk4-mk3-9_465702

**Comment by jhenley01** [comment]
I've had this issue multiple times as well. Sometimes wiggling filament will fix it, sometimes not. Between this and the thermistor issues the MK4 has been disappointing. For the price of these machines compared to the competition you'd think they could get basic things like a filament runout sensor to work correctly.

**Comment by PhLacoude** [comment]
I just experienced this issue for the first time after running out of PLA filament 1h20 minutes into a 1h51m print. This is VERY annoying when a simple GUI change could override the sensor.

**Comment by bartbergman** [comment]
I had the same issue yesterday.
It's definitely a hardware issue (as already acknowledged by Prusa).
I had to manually clean the sensor because I didn't have compressed air available, but a tip from Prusa Support was to try to wiggle the metal ball inside the sensor with a magnet to unblock it. Haven't tried that yet, but will definitely try that if this occurs again.

**Comment by n3wtron** [comment]
It happened also to me and I solved (that time) brutally disconnecting the printer from electricity, inserting the new filament in the extruder opening the "blocker", reconnecting the printer and finally clicking "change filament".

It worked, I don't know if it will always work but...

**Comment by Prusa-Support** [comment]
The main cause is most likely debris around the filament sensor and this may affect pretty much all types of printers and filament sensors we tested so far, in various ways, and to different extents.
Of course, there may be specific cases where we can't exclude other causes of failure - check with Customer Support -  but again, it is usually a matter of debris. 

The first suggestion should probably be keeping the filament dry as moisture makes the filament more stringy and brittle.
Also, inspecting and cleaning the extruder gear and its surroundings is a simple practice that is often overlooked.
https://help.prusa3d.com/article/regular-printer-maintenance-mk4_419000

Disconnecting components while the printer is running is something to be _avoided_.
Unplugging the printer should not be necessary and may lead to complications. It may make the printer simply ignore the filament sensor status and encourage the customer to continue without solving the problem itself. I don't feel like recommending this method either.

Removing and inserting the filament a few times could knock off debris from the filament sensor, otherwise, you can clean the filament sensor with canned air, or take the sensor's ball holder off to proceed with cleaning.
https://help.prusa3d.com/article/filament-sensor-mk4-mk3-9_465702
Consider trying cleaning with compressed air _from below as well_ (through the idler door) but removing the ball holder for cleaning would be more effective.

Michele Moramarco
Prusa Research

**Comment by pigskins65** [comment]
I have to respectfully disagree, at least with the firmware level I am at.
People have proven that they can get this issue to occur on demand. In my
case it happened every time I ran out of filament mid-print. I had to shut
off the sensor and pay close attention to the amount of filament remaining.

**Comment by PhLacoude** [comment]
I agree with @pigskins65. Could you simply add a button in the GUI that would allow us to override the faulty detection of the filament removal?

Just to recap, the filament is removed BUT even if you reply YES to the "Was filament removed?" the printer KEEPS asking you to remove filament. At this point, it would be very simple to add a button say would say "Are you REALLY sure the filament has been removed?". It would let us go on with our lives.

In my case, I tinkered with the wire of the filament sensor, making sure it would stop indicating there was filament left when there weren't. But this is a hardware solution to what should be a software issue.

**Comment by jhenley01** [comment]
I’m not really buying the “this may affect pretty much all types of printers and filament sensors” answer. 

I’ve owned several printers from different manufacturers and the MK4 is the only one I’ve had repeatable failures with the filament runout sensor. 

Simply admitting you guys made a mistake with the design and working on a fix will go a lot farther with your customers than trying to claim it’s a normal thing that happens with all printers.

**Comment by fkirill** [comment]
+1, same problem here. THe filament ran out, the printer is stuck with "Please remove the filament" message and there is nothing I can do. I will have to start the print from scratch.
I must stay this I have already ran out of filmant multiple times and everything worked fine. So maybe this is a more complex issue that only occurs sporadically, maybe some kind of race condition etc.

**Comment by miyconst** [comment]
Today my MK4 got stuck at the same annoying message, in my case it turned out that the filament end was bent and it got stuck in the path, from the outside it looked empty but with a torch light, I saw the issue. Used a needle to straighten the string and pull it out, then I was able to resume the print.

Still, **the current behavior of the firmware/printer is simply not acceptable**.

- There shall be at least two buttons: `[Purge more filament]` and `[Try unloading again]`.
- Ultimately, it should be possible to save the print progress, reset the printer, and continue afterward.

**Comment by pjvenda** [comment]
Had this issue triggered mid-print when a filament ran out. Current firmware is 5.1.2.

Was asked to check the filament had been removed (no filament was visible - it had been pulled into the print), confirmed, then got stuck in the "Changing filament' - 0% screen. Had to reset the printer and loose the print.

After resetting and inserting new filament, a large amount of fairly thick filament came out of the nozzle, as if too much had been pulled in... nonetheless, new filament was loaded successfully.

**Comment by jascha-sacc** [comment]
+1 to all this!  I just lost a print twice because of this problem (the printer's in a shared space, the spool had some nasty tangles in it that didn't show up until partway through, and that's why I was changing the filament). 

Specs:
- Prusa MK4 upgraded from mk3S, but has been working fine for weeks
- Firmware 5.1.3

*Absolutely* having a GUI override would make sense here.

**Comment by pjvenda** [comment]
> Had this issue triggered mid-print when a filament ran out. Current firmware is 5.1.2.
> 
> Was asked to check the filament had been removed (no filament was visible - it had been pulled into the print), confirmed, then got stuck in the "Changing filament' - 0% screen. Had to reset the printer and loose the print.
> 
> After resetting and inserting new filament, a large amount of fairly thick filament came out of the nozzle, as if too much had been pulled in... nonetheless, new filament was loaded successfully.

I should add that the exact same situation happened again the same day. This time a small amount of filament was left on the printing head. I tried the technique described around using a piece of filament and poking it in the hole repeatedly, which after some insistence actually worked.

The sensor does not appear dirty or malfunctioning because it correctly detects the presence of filament when swapping filaments while a print is not running. My suspicion is that the path of the filament is not clearing enough to return the sensor detector because when poking the hole with the filament I could not feel the springy resistance of the ball as I did when replacing the filament outside the printing operation.

I hope this helps. And I hope somebody is working to fix this.

**Comment by bolsoncerrado** [comment]
The very few interactions "Prusa" has done in this thread suggest otherwise.

**Comment by jascha-sacc** [comment]
Had a long chat, actually - maybe it'll make a difference!

**Comment by PhLacoude** [comment]
> 
> So definetly needs a firmware revamp on all these issues starting with MORE options on screen and defiently powerpanic NOT resuming correctly when youre changing filament...
> 

I strongly agree with "MORE options on screen". This whole thing could be circumvented with additional screen options. I understand the hardware aspect might be difficult to sort out. But there is no excuse for not adding a couple more buttons on the screen.

**Comment by jhenley01** [comment]
Another day and another failed print at 98% complete because of the runout sensor issue. 

I'm also running 6.0 RC1 firmware and can confirm the issue is still there and still no menu option to override. Seems to me that Prusa really has no intention of fixing this issue. Pathetic.

The really dumb part of of this is when you do a filament change it asks you if unload was successful. Why the heck don't they have that on filament runout as well?????

**Comment by bolsoncerrado** [comment]
Perhaps as easy as "Ignore filament sensor for the rest of the print" thing or option to be pressed available during a print and filament change or filament runout or something like this besides the retry/cancel requests...

It's sad Prusa guys are no longer sharing further insights here....

**Comment by bolsoncerrado** [comment]
GEez, v6.00 ahead and no word for sensor improvement. Awesome :(
https://github.com/prusa3d/Prusa-Firmware-Buddy/releases/tag/v6.0.0-RC1

**Comment by jascha-sacc** [comment]
So I had a great convo with a Prusa help person via chat where I dropped the URL to this thread - let's have more people also do that and keep the issue fresh in their minds.

**Comment by unclej84** [comment]
After having modified the sensor and idler for the mmu3 I had no further issues with this. Unfortunately the seem not to sell a kit or spare parts for this modification without the mmu3 and the assembly manual still shows the old ball/magnet/spring and idler.

**Comment by bolsoncerrado** [comment]
That would clearly prove their really **BAD** hardware decision on this...

**Comment by jhenley01** [comment]
I’m betting they’ll do one of two things. 

They’ll either quietly do a hardware fix for new units going out and try to avoid having to replace parts for existing customers. 

Or they’ll do what they did when the Mini had all heat issues with the MINDA probes. They came out with the “SuperPINDA” kit and expected people to buy an “upgrade” to fix their design flaw.

**Comment by Prusa-Support** [comment]
Unfortunately, our reply hasn't changed - https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3129#issuecomment-1944407797 - but rest assured that your comments are not being ignored.

As explained in the similar/same issue before this one - https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/2970#issuecomment-1924886097 in fact opened before the release of Prusa MK4 but also listing MK4 cases -  similar problems may be observed _to some extent_ with pretty much all kind of filament sensors we _tested_ and/or released so far, and, as many highlighted in this thread, the problem seems to be only reproducible in the case of mechanical blockage.

There are chances that the [_stamping_ process](https://blog.prusa3d.com/major-firmware-release-6-0-0-prusaslicer-2-7-4-and-prusa-connect-updates-summary_95627/) introduced in [Prusa Slicer 2.7.3](https://github.com/prusa3d/PrusaSlicer/releases/tag/version_2.7.3-alpha1) will reduce some scenarios _where the filament conditions and settings are the potential problems_.

However, as per https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/2970#issuecomment-2016865010, our best recommendations at the moment remains the following.

- Remove and insert the filament multiple times in an attempt to knock off debris and realign the filament sensor ball. 
- Blow canned air from [the hole above the filament sensor](https://cdn.help.prusa3d.com/wp-content/uploads/2023/08/5ac2eee63621ff1e9030242fe03fa27c.jpg).
- Blow canned air from below, through [the opened idler door](https://cdn.help.prusa3d.com/wp-content/uploads/2023/03/XL_Extruder_Open2.jpg).
- Remove the Planetary Gear parts "[like a sandwich](https://cdn.help.prusa3d.com/wp-content/uploads/2023/10/67c593556671a61a79c52036477d3454.jpg)" to prevent gear slippage. Hence access, remove, clean, and realign filament sensor ball & spring. Repeating the [Gearbox alignment](https://help.prusa3d.com/article/gearbox-alignment-mk4_465779) is very recommended after this operation.
- In the case of a recurrent issue please consider
  - drying filament or comparing the recurrence of the problem with other filaments,
  - repeating the Filament sensor calibration ([_Calibrations & tests_](https://help.prusa3d.com/article/lcd-menu-mk4_413171) menu).

Basic guidelines can be found in the relevant Help article linked below.
https://help.prusa3d.com/article/filament-sensor-mk4-mk3-9-xl_465702

Michele Moramarco
Prusa Research

**Comment by bolsoncerrado** [comment]
How sad to read...

**Comment by PhLacoude** [comment]
For the record, I strongly disagree with Michele's reply.

When the spool runs out of filament, and the filament sensor correctly senses it, there is no reason to keep the user in an infinite loop asking him/her to remove the filament after it has already been removed.

At a minimum, instead of perpetually asking to remove the filament that is no longer present, the menu on the screen should let the user CONFIRM that the filament has been removed, and let the user proceed at his/her own risk. From experience, we know if we have filament in our machines.

FOR ALL PEOPLE WHO WOULD COME TO THIS PAGE LOOKING FOR A WAY OUT IN THE MIDDLE OF A 10+ HOUR PRINT, you can fix the issue by unplugging the filament sensor on the LoveBoard. The MK4 will detect that the filament is gone and let you out of this endless menu. You then plug the filament sensor back. Cumbersome but it works. https://help.prusa3d.com/guide/5-nextruder-assembly_434014#435286

**Comment by bolsoncerrado** [comment]
Not only I agree with the software solution of a confirm question but I also have the feeling that upon unloading midprint, somehow the sensor calibration value may go nuts, that's the only explanation on why moving in and out in quickly succesion the filament triggers the response BECAUSE WHEN THERE'S NO PRINT GOING ON THERES NO SUCH ISSUE WHEN UNLOADING FILAMENT.

So it must be something firmware glitch side. A variable not correctly called or emptied or whatever.

Anyhow the confirmation should be more than enough and not leaving you option less with just a "please remove" when you have already done so mid print!

Prusa response to this is just lame.


For the record, I have the feeling unplugging and plugging back live components MAY FRY something so beware!

**Comment by Prusa-Support** [comment]
Disconnecting components while the printer is powered on is something you should really avoid. Not to mention that this workaround would most likely only postpone the problem.
https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3129#issuecomment-1944407797

Cutting power to trigger an intentional Power Panic and resume later, in this case, is not an intended feature - results are not guaranteed. I guess it may work in some cases, but again it is not a solution and will most likely only postpone the problem.
https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/2970#issuecomment-2016865010

If the ultimate request is to have an option on the screen to postpone the problem troubleshooting, I can't assure a positive outcome but it will be my concern to collect all feedback about it and share it with the developers.
For now, our recommendation remains to clean and inspect the filament sensor.
https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3129#issuecomment-2105938590

Michele Moramarco
Prusa Research

**Comment by mobilla** [comment]
I agree disconnecting components is a bad idea mid print, and relying on power panic is also a bad Idea. However they are the ONLY options right now because a "skip check" button isn't available. If I am 20 hours into a 21 hour print and need to change filament I am NOT going to cancel the print and start again with a larger spool when a simple button will do. I would rather delay that problem, and try to fix it in an hour, after the print is done.

Until the real issue can be fixed, please implement the simple solution of a skip check button so we can stop needing to do riskier things which could result in real damage or even fire if you accidentally unplug something like the heater element. I would rather risk a failed print than waste 20 hours or break my printer completely.

**Comment by miyconst** [comment]
@Prusa-Support 

I love my Prusa MK4, I plan to buy an XL, and I want to love Prusa as a company, but this issue makes me doubt the competence of people answering here.

What I am going to write will most likely be considered by some as rude and unprofessional, so I say upfront that I don't want to hurt or target anyone, I want Prusa to improve.

- All the official responses sound like the company is tone-deaf and comes up with all sorts of excuses instead of addressing the real issue.
- The suggested and/or official "take on the issue" does not in any way help people in the situation described.
- Nobody asked Prusa to adhere to an unsafe solution.
- In most cases, the problem can be easily solved by retrying the already existing procedure of unloading and then reloading the filament.
- In my particular case, the problem could have been solved by asking the printer to retract the filament so I could take it out from the Nextruder instead of trying to hook it with a needle and drag it out with force.
- Many people report and ask for the same thing: give us the option to restart the procedure, drain the filament, retract the filament, and finally just ignore the sensor and continue printing instead of the stupid "Remove the filament" screen with no other options.
- Yes, "ignore and continue" is postponing the issue, it's not solving it. And yes, that is what we want! When you have a 20-hour print with 2 hours left, you do not want to fiddle with the printer, you want to finish the print and then figure out what went wrong.
- The current experience can be compared to a car that stops on a highway due to a faulty fuel level sensor. The fuel is there, the engine starts, the car runs just fine, only the sensor is faulty, yet the manufacturer declares that the only way to solve the problem is to get towed to a workshop.

**Comment by bolsoncerrado** [comment]
A-Men!

**Comment by PhLacoude** [comment]
"Disconnecting components while the printer is powered on is something you should really avoid. Not to mention that this workaround would most likely only postpone the problem."
> Yes, @Prusa-Support, I strongly agree that this is something we should really avoid.

But if you gave us an "ignore and continue" button, as @miyconst suggest, we could finish the print after the spool ran out on a long print. I believe that none of us have proposed this change to do 20+ filament changes. Most of us would have liked to have this option to use once to complete a long print after a spool of filament ran out.

It is not even always true that it postpones a problem. As many of us said, there was nothing in the way of the filament and, after resetting the machine, the filament sensor would no longer "see" a filament (or debris).

**Comment by otaviojr** [comment]
+1

I'm facing the same problem here with the new MMU3 and the new idler filament sensor. 

A 2-hour, single filament print stopped five times.

I need to click continue after it ejects the filament, and it will reload it and continue printing.

Annoying. I need to be there to click a button.

**Comment by rwillett1** [comment]
My Prusa Mk4 arrived a few hours ago, set it up, got it working, decided to simulate an out of filament issue and I have the same sodding problem as everybody else. If I push an old piece of filament in 5-10x, it appears to work. 

So I can safely say that Prusa has not quietly fixed the hardware problem or used software to fix it. This thread started 11 months ago and we still have it on brand new printers. Mine is three hours old.

I've just brought a brand new printer that is not as good as my old Mk3 that it replaces and *was* going to be given to the local school. 

I am furious that this major issue has not been fixed. I'm contacting Prusa to send it back, its not fit for purpose.

Rob

**Comment by bolsoncerrado** [comment]
I'd really like to see Prusa staff to come here one last time and repeat, after this last evidence, that it's filament debris the ONLY issue this filament sensor has...

As the saying goes... "There's none so blind as he who will not see." 

Geez.

**Comment by rwillett1** [comment]
I've just cut the filament to see what happens. 

Filament unloaded OK, this time it did load but I had to reload it to get it to work. it put up the message "Ramming" on the screen. I pushed a little harder and it worked. 

So lets do another one and see what happens. This is on a printer that has less than two hours usage from coming out the box.

The last two have worked OK, lets keep doing it and see.

Rob

**Comment by otaviojr** [comment]
> I've just cut the filament to see what happens.
> 
> Filament unloaded OK, this time it did load but I had to reload it to get it to work. it put up the message "Ramming" on the screen. I pushed a little harder and it worked.

I never cut the filament. In my case I just click continue and it will reload and continue to print. No interaction with the printer.

I will try to calibrate the finda better.

This never happened with mk4 alone. It started after the MMU3.

**Comment by rwillett1** [comment]
The filament sensor just failed to recognise it was out of filament. Six hours of printing down the drain. If this was a Creality Ender 3 Pro V1 from five years ago, fair enough, this is a £1,100 machine and it doesn't even recognise it's out of filament. Its three days old.

Just tried it again and its happily printing with no filament. This is garbage. 

Also Prusa do not have a Live 24/7 chat, isn't working at the moment.

**Comment by otaviojr** [comment]
I did calibrate my Finda sensor and was able to print a 5h part with no issue.

I believe my problem is not related to this.

I think it was the Finda all the time.

**Comment by MikeDabrowski** [comment]
Happened to me second time. First time it was on last 5v and this time is on latest 6.0.2. Old filament unloaded - asked to remove the end, tried to load to finda twice and just sits there waiting for idk what.

Last time I talked with support they couldnt help it and we threw it off as a random bug. This time I am waiting 20min for the chat. Already restarted the mmu3 twice, on this latest version there is an option now to disable MM.

**Comment by bolsoncerrado** [comment]
At this point im not sure but seems like MMU bug is "slightly" different than stock MK4 bug... could be same bug or just different firmware bug, but both could be solved with a CANCEL or GO ON options on the menu... *sigh*

**Comment by Thf772** [comment]
I've had the same issue happen on an Mk4 I built just a few days ago, using firmware 6.0.3+14092. Given the behavior I've observed, I think it's a mix of hardware issues and software bugs (or if not a bug, behavior one would not expect with common sense alone).

Hardware issues: when I have the block (printer asking to remove filament when no filament is present), manipulating the filament sensor ball with a piece of filament or the 1.5mm allen key unblocks the printer, but not instantly. After enough manipulations, I can clearly feel resistance when inserting a piece of filament, and by shining a smartphone light through the open idler and up the filament path, I can clearly see the ball extending into the filament path. A few seconds later *without touching anything*, the printer manages to move on to the next step in the change process. My best guess is this comes from the sensor taking time to react to the change in the magnet's presence.

Software issue: in order to avoid the block, the procedure I have found works all the time is to remove the filament only *after* confirming it has been removed successfully, and only once the printer actually asks to remove it. I don't understand why such a procedure is necessary: if the printer asks the user to confirm the filament has been removed, the first thing anyone would do is take the small bit of filament out of the extruder and *only then* confirm that yes, they were able to take it out of the printer without any issue. Asking to remove it when it has already been removed is counter-intuitive to the extreme, especially since the printer *knows* whether the filament is still present or not.

I don't know about the hardware issue, if anyone can confirm and refine the cause it would be good. As for the software part, I don't know if it's an actual bug or a counter-intuitive feature, but I suggest changing the filament swap procedure so that it skips the "Please remove the filament" step if the filament sensor already reports no filament being present.

**Comment by paulcobbaut** [comment]
>     * Many people report and ask for the same thing: give us the option to restart the procedure, drain the filament, retract the filament, and finally just ignore the sensor and continue printing instead of the stupid "Remove the filament" screen with no other options.

This! Please!
I have this issue in about one in twenty colour changes, even today with the latest MK4 firmware.

**Comment by darkmattercoder** [comment]
Happened today with one extruder of my XL. Since the firmware had no button to force-confirm, I cut the power and the print finished successfully.

Please add a force skip check button.

**Comment by rwillett1** [comment]
I still get this on both my MK4's. The printer keeps thinking there is filament there when I have removed it.

I also get this sometimes, but less often, when it runs out of filament and the printer keeps on printing.  Thats really annoying.

Also the current Prusa firmware doesn't communicate correctly (or at all) with Octoprint when it is out of filament. This has now been fixed with an Octoprint plugin here, https://github.com/jneilliii/OctoPrint-PrusaFilamentRunoutMonitor, so thats one less thing to worry about. It is not quite as good as the developer and I would like (I funded the work), but after talking to Ginna, it's as good as it can be given that Prusa don't appear to give a rats arse about this problem.

**Comment by bolsoncerrado** [comment]
IRONIC MODE ON

Stop denying this bug! This does not exist!

Sincerely, 

Prusa Support. 

IRONIC MODE OFF

**Comment by mvayala** [comment]
Have had this issue 4-5 times in a month. @Prusa-Support stop wasting people's time with this asshole design and put in a simple option

**Comment by jascha-sacc** [comment]
Yeah - this is really bad of them.  I had two online chats with support
last year where I really thought I'd gotten through and put it on their
to-do list!  But alas, I guess not?  Poor showing, Prusa.

-j-


- - - - - - - - - -
Jascha Narveson
computer department
Saint Ann's School

**Comment by abatkin** [comment]
I discovered that my filament runout sensor was mis-calibrated, and it (almost) always thought that filament was inserted. In normal operation (even for normal filament changes that aren't happening mid-print) this isn't an issue, and you won't really even notice that this is the case. I then re-calibrated the filament runout sensor, and I haven't had an issue since.

Actually, one time I still had an issue when I pulled out the filament and some stringing was left behind, but once I jammed a piece of filament in a few times to move it out of the way, it cleared.

So here's my suggestion: Go to the calibration menu, try re-calibrating the runout sensor, then insert/remove filament a few times (doesn't have to be a full load/unload, i.e. you can do this cold) and make sure it properly and reliably recognizes that the filament is being inserted and (more importantly) removed.

I DO think that Prusa should still provide an option on the menu when doing an in-print filament change (whether it's due to a planned color change or an actual filament runout) to "continue" past the "remove filament" screen even if the sensor _thinks_ that filament is still inserted. Even if it has a big red warning.

**Comment by jascha-sacc** [comment]
Yeah - I'd be happy with the "Proceed at your own risk" style warning on
the UI - I'm happy to take responsibility for overriding the system if the
system clearly sometimes makes mistakes!

-j-


- - - - - - - - - -
Jascha Narveson
computer department
Saint Ann's School

**Comment by bolsoncerrado** [comment]
I was on the same boat of miss calibration twice but the problem returned shortly after so I'm not sure if it's their way of calculating or the margins they use to realize filament is in or not but what it makes no sense and no logic is to give you NO OPTION to cancel nor continue if filament is out and printer keeps requesting you to remove it. That's plainly DUMB at this point in the... "market".

**Comment by ZeekWorld** [comment]
Found this long thread, same issue on the Prusa Mini, color change, remove filament, then stuck at the screen when there is no filament.
Happened three times today, I don't usually print with color change steps but today got three unfinished prints.
Hope for a fix.

**Comment by knowtheory** [comment]
> Happened today with one extruder of my XL. Since the firmware had no button to force-confirm, I cut the power and the print finished successfully.
> 
> Please add a force skip check button.

Thanks for this suggestion.  My mk4 just ran out and insisted i needed to remove non-existent filament.  Cutting the power and having the print restart did the trick for me too.

**Comment by fuzzievohs** [comment]
I've been having the same issues.  Tried the suggested fixes.  Only a reset fixes it, which ruins the print.  On reset, the filament loads without issue.

**Comment by bolsoncerrado** [comment]
Is there any reason to keep this issue open then? Legit question.

Furthermore it seems clear you don't give a damn at hardware level since MK4S didn't introduce any fix to this lame sensor solution so there you go Prusa, another pissed customer leaving the boat.

You can close the issue now.

**Comment by darkmattercoder** [comment]
The issue should stay open so that it is clearly visible that prusas simply is refusing to give us a manual override possibility.

**Comment by jhenley01** [comment]
Lovely that is the kind of support you get when you spend double the money for what is supposed to be a "premium product."

**Comment by PhLacoude** [comment]
The issue should stay open because it has not been fixed. I have difficulty understanding why a simple manual override (which would represent an exceedingly small amount of code) cannot be implemented. I think the Prusa representative on this feed has crossed the fine line between reasonable stubbornness and passive-aggressive rudeness.

**Comment by Ludicrous2k** [comment]
I found a work around!!!

After you pull out your old filament, stick your new filament in as far as it'll go, and then pull the power cable. Once you plug it back in and and the printer boots up you can then select resume print and it'll start auto feeding the new filament and continue where it left off. LOL

**Comment by bolsoncerrado** [comment]
Sounds risky as hell haha but if it works for you.... 🤣

**Comment by bolsoncerrado** [comment]
> However the fact a power cycle resets it, tells me that there is a bug somewhere as it should not work. If there was debris in before (as Prusa support keep holding onto this figment of truth) and keep repeating that, then after a power cycle it should still be there.

Ironically, this has been in the first OP since the beginning of this BUG post... So yeah, I guess it's a matter of pride on Prusa side now... :/

**Comment by micro17** [comment]
Sorry to hear that prusa is not willing to investigate this with the given respect and openness.

This problem seems really problematic to diagnose and I understand that.
Im also willing to accept that giving us an option on screen to skip it is not the right solution.

But Prusa gave us nothing at all with this issue.
Perhaps showing the sensor value to diagnose better, saying we know the issue and the solution is re inserting filament until sensor is reset or sth.

But nothing. Just Support being really frustrated with us instead of taking the side of the end users with the Hardware department.

I mean I respect people showing their frustration openly but here we as users are clearly not the right ones to address.

**Comment by froboy** [comment]
I ran into the same problem on a mini+ and this was the issue: https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/2970#issuecomment-2015804424

If you pull the filament out before telling it that it was successfully removed, then you'll be stuck in an un-escapable sensor warning.

**Comment by bolsoncerrado** [comment]
Definetly a GUI issue no one's willing to address...

**Comment by froboy** [comment]
![image](https://github.com/user-attachments/assets/85fc7a0c-527f-4fa4-abca-b6bda3baa43a)

This screen should say "Do not touch the filament yet!!! If you already pulled the filament out put it back in. If you push yes while the filament is out then your print will be dead. Srsly."

**Comment by bolsoncerrado** [comment]
Something as simple as "CONTINUE" would do actually.... :(

**Comment by dolezelk** [comment]
Prusa is no longer Fixing bugs which are present many years in the firware, they are just focusing to release new printers.
What a crappy printer Mk4s

**Comment by picodexter** [comment]
Found a workaround for myself.

My attempts for an XL toolhead:

- repeatedly inserted and withdrew a short piece of filament (cut at an angle) at a slow and fast (4-5 per second?) pace, even slowly rotating 360° to make sure all sides of the walls were covered, in case something was stuck there and the filament would help clear it out. With idler door closed and open. No success
- compressed air, with idler door closed and opened. No success
- after compressed air again inserting / withdrawing the piece of filament. No success

What worked:

- took a 2 mm Allen key and inserted it, but not far enough to hit the gears (probably want to keep idler door open to make sure). You should clearly feel the steel ball in front of the filament sensor being pushed out of the way, actuating the sensor

For some reason when the Allen key was inserted, the display switched to the screen with the Continue button, and when I took it out again, it went back to the "Please remove the filament from the extruder" without any buttons. It seemed to have the sensor states flipped around somehow.

I then pushed the Allen key back in far enough to get back to the screen with the "Continue" button, hit the button, pulled the Allen key back out, connected the PTFE tube and pushed the filament in like usual.

My assumption is that the filament sensor state will recover after the print or a restart, so it was probably a fluke.

Hope it helps someone else

**Comment by rwillett1** [comment]
I've tried that before in the past, just tried it now as a test and it didn't work for me. Tried it twice and the Prusa happily started printing with no filament in. 

At the end of the day Prusa took a simple but highly effective ball and magnet in the MK3 and turned it into a complex, awkward and temperamental system in the MK4. 

I **suspect** this is caused by micro particles of plastic building up in the ball bearing channel and this is stopping the ball bearing from triggering whatever Hall Effect system is now in place. This might have saved 10p or cost 10p but it won't be much either way. 

I also **suspect** that the new motherboard can't handle any other system so they can't easily change it without a big hit on the bottom line.

I **know** that nobody from Prusa takes this seriously as they've done sweet FA about it beyond occasionally coming on here, wringing their hand piously and saying they can't find anything wrong. 

Too many good people who print a lot are logging this issue. I am an average printer without some of the know-how they have, so I can only imagine how they feel doing the volumes they do. I only do a few rolls a week so can live with the hassle.

However these are my last two Prusa's. If it wasn't for Bambu being even bigger dickheads than Prusa, difficult to imagine I'll grant you,  I'd have flogged mine at a loss and moved to Bambu.  However I keep a very close eye on anything that could replace the Mk4's as I'm tired of the hassle with them,

As Prusa are incapable of finding this issue internally, feel free to ship me a MK4 and when it happens on my printer I'll ship mine back and ship it to you at no cost to you. You can then do what you like with it and I'm no worse off than before. You keep mine and I'll keep yours.

Rob

**Comment by bolsoncerrado** [comment]
...not to mention that ridiculous "gravity hole" that wont make any good to dust/particle buildup "INSIDE" the sensor area.... JC....

**Comment by rwillett1** [comment]
Perhaps Prusa have never heard of electrostatic attraction?

**Comment by mvayala** [comment]
Remains a major issue with no actual solution or support. Fuck this company.

**Comment by CZDanol** [comment]
That's not true. Yes, we were swamped by the CORE one deadlines and were focusing heavily on that for the last several months, but that doesn't mean that we don't care about our other printers. We are now slowly getting back on track and spreading our attention more. Plus, it's not like our C1 development wouldn't benefit other guys as well. We have done improvements and bugfixes that all printers benefit from.

**Comment by bolsoncerrado** [comment]
Well this "issue" will be soon 2 years old and you havent added even a miserable "CONTINUE" button to the software, so I guess thats how much you care about it...

**Comment by bolsoncerrado** [comment]
having posted that same sentence "several months ago" would have helped a bit...several months ago. Now we feel abandoned.

**Comment by MikeDabrowski** [comment]
They said that flashing custom fw is now not breaking warranty so maybe we
should work it out ourselves

**Comment by CZDanol** [comment]
You can also submit a pull request :)

**Comment by jhenley01** [comment]
> Just to remind folks, you're ragging on the maintainers of an _open source repository_. If you feel strongly about this issue you should ask how to help, not just berate the maintainers, please.

Why on earth should we be expected to help fix an obvious issue on a $1000 printer we paid for? 

Also there have been many attempts to help fix the issue in this thread but Prusa won't lift a finger on their end.

**Comment by mvayala** [comment]
It's abundantly clear where this company's priorities are. Canceled C1 order and linked this thread. This isn't a "nice to have," it's a normal function that fully breaks any ability to refill material during a print. "Open source" in this case seems to mean "let the community fix problems with the commercial product we sold them, we'll ignore it until there's a fix and implement it later," maybe something lost in translation? 🤷‍♂️

**Comment by bolsoncerrado** [comment]
It's not only the ability to swap filaments. In my case are ruined prints because filament sensor didn't kick in when filament ran out...

I'm also done with the company. Don't get me wrong. I was a big fan and bought every single release they made even the foolish anniversary editions of their stuff. I know it must be hard to compete with current global situation and market opportunities, but "open source" is no longer "a marketing thing".

How many "open source" buddy¦cheese¦lasagna boards" can you find on the open market community??

**Comment by Tupson444** [comment]
Happened to me today. Filament ran out and the printer paused as expected, but when I removed the filament it kept asking to remove the filament, without an option to continue. I did manage to make it work by inserting and removing filament multiple times, but if the sensor got really stuck I would lose the 10h print.

Yes, if this happens, the filament sensor should be cleaned to prevent such problem in future prints, but why we have to lose the current print for not having the option to just disable the sensor for the rest of the print and finish printing?!?! Even if the hardware or filament ramming gets improved, some sort of malfunction of the sensor may happen, so there should be an override. Even if it is extremely rare thing to happen, there should be an override option (there could be a warning shown on the screen that the sensor should be checked before starting more prints).

Focusing on the fact that something else should be done to fully fix the problem and ignoring that the current print could be saved by simply disabling the sensor is just flawed thinking.

**Comment by MikeDabrowski** [comment]
@Tupson444 I don't remember if I ever described the semi workaround I found so here it goes

You have to pull the plug so the printer experiences power loss. This prompts it to resume the work and somehow ignores the loading loop. The caveat is that you have to time it right. You have to remove the filament and insert new one all the way in up to the nozzle. And also pay attention not to push out melted one from the nozzle doing that. This is because you want to avoid gaps between layers. After power loss the printer does not prime the nozzle but just continues where it left.

This method might require some refinement. But the alternative is to scrap the print so it is worth considering.

Edit; @Tupson444  - what firmware were you running when this happened?

**Comment by Tupson444** [comment]
Yea, we have to use risky workarounds because Prusa failed to recognize the importance of an override (there is an option to disable the sensor in the menu, so why not have that option during the filament change procedure). I first considered to unplug the sensor (I see that others have suggested that too), but I didn't have to. I'll keep in mind your suggestion too.

I'm using 6.2.2, but I haven't seen anything related to this in release notes for newer versions.

**Comment by Prusa-Support** [comment]
The solution to this problem - **besides cleaning the filament sensor** - should be the implementation of an option to _skip the filament loading failure_ (<ins>temporary filament sensor deactivation</ins>) as requested by the community.

The feature is technically already active on firmware series 6.3.x, which is currently already running on Prusa CORE One, and unfortunately, not backwards compatible with firmware series 6.2.x that is currently still running on Prusa MK4/XL.

A new firmware series 6.4.x will merge the old MK4/XL and new CO firmware, and port this implementation to the previous printer models. 

Michele Moramarco
Prusa Research

**Comment by PhLacoude** [comment]
> The solution to this problem - **besides cleaning the filament sensor** - should be the implementation of an option to _skip the filament loading failure_ (temporary filament sensor deactivation) as requested by the community.
> 
> The feature is technically already active on firmware series 6.3.x, which is currently already running on Prusa CORE One, and unfortunately, not backwards compatible with firmware series 6.2.x that is currently still running on Prusa MK4/XL.
> 
> A new firmware series 6.4.x will merge the old MK4/XL and new CO firmware, and port this implementation to the previous printer models.
> 
> Michele Moramarco Prusa Research

This is great news, even if it comes much later rather than sooner (and after you told us herein this was not an issue.) Thank you!
